frontend/admin/src/js/components/user/UserProfilePrintButton.tsx

```diff
@@ -0,0 +1,73 @@
+import { Button } from "@common/components";
+import { commonMessages } from "@common/messages";
+import React, { useRef } from "react";
+import { useIntl } from "react-intl";
+import { useReactToPrint } from "react-to-print";
+import printStyles from "@common/constants/printStyles";
+import NotFound from "@common/components/NotFound";
+import Pending from "@common/components/Pending";
+import { Scalars, useGetUserProfileQuery } from "../../api/generated";
+import UserProfileDocument from "./UserProfileDocument";
+
+export interface UserProfilePrintButtonProps {
+  userId: Scalars["ID"];
+}
+
+export const UserProfilePrintButton: React.FunctionComponent<{
+  userId: Scalars["ID"];
+}> = ({ userId, children }) => {
+  const intl = useIntl();
+
+  // would be nice to only fire this if the button was clicked but you would need an imperative version of the query that you could await on
+  const [{ data: initialData, fetching, error }] = useGetUserProfileQuery({
+    variables: { id: userId },
+  });
+
+  const componentRef = useRef(null);
+  const handlePrint = useReactToPrint({
+    content: () => componentRef.current,
+    pageStyle: printStyles,
+    documentTitle: "Candidate Profile",
+  });
+
+  const userData = initialData;
+
+  if (fetching) return <p>{intl.formatMessage(commonMessages.loadingTitle)}</p>;
+  if (error)
+    return (
+      <p>
+        {intl.formatMessage(commonMessages.loadingError)}
+        {error.message}
+      </p>
+    );
```

Comment:
With the new `<Pending />` component, this can safely be removed.

Reply:
> Maybe we can add a prop to the pending component to allow it to be rendered inline rather than full screen. Also, I'm not seeing the background colour for that component anymore

This is just me making notes to myself. Feel free to ignore if you don't care 😅